src/utils/utils.cairo

```diff
@@ -514,4 +514,47 @@ namespace Helpers {
             return a;
         }
     }
+
+    // @notice convert bytes to little endian
+    func bytes_to_byte8_little_endian{range_check_ptr}(
+        bytes_len: felt,
+        bytes: felt*,
+        index: felt,
+        size: felt,
+        byte8: felt,
+        byte8_shift: felt,
+        dest: felt*,
+        dest_index: felt,
+    ) {
+        alloc_locals;
+        if (index == size) {
+            return ();
+        }
+
+        local current_byte;
+        let out_of_bound = is_le(a=bytes_len, b=index);
+        if (out_of_bound != FALSE) {
+            current_byte = 0;
+        } else {
+            assert current_byte = [bytes + index];
+        }
+
+        let (bit_shift) = pow(256, byte8_shift);
```

Comment:
it may be better to do like in `pow256_rev`, ie using a `dw` array

Reply:
I don't know how we could use a `dw` array since `byte8_shift` is maximized by `bytes_len` value

Reply:
I think that `0 <= byte8_shift < 8`, see:

```
let byte8_full = is_le(a=7, b=byte8_shift);
let end_of_loop = is_le(size, index + 1);
let write_to_dest = is_le(1, byte8_full + end_of_loop);
if (write_to_dest != FALSE) {
// ...
// byte8_full is reinitialized
```

Reply:
Ok got it, thanks!